api-presigned.go

```diff
@@ -45,33 +35,18 @@ func (c Client) presignURL(method string, bucketName string, objectName string,
 	if err := s3utils.CheckValidBucketName(bucketName); err != nil {
 		return nil, err
 	}
-	if err := s3utils.CheckValidObjectName(objectName); err != nil {
-		return nil, err
-	}
```

Comment:
why should object name validation be relaxed?

Reply:
To make it generic, need to move this check to presignedGetObject()